bouncer/shared/utils.ts

```diff
@@ -977,3 +977,30 @@ export function calculateFeeWithBps(fineAmount: bigint, bps: number): bigint {
   const divisor = BigInt(10000 / bps);
   return fineAmount / divisor + (fineAmount % divisor > divisor / 2n ? 1n : 0n);
 }
+
+export function tryUntilSuccess(
+  closure: () => Promise<boolean>,
+  pollTime: number,
+  maxAttempts: number,
+): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    let attempts = 0;
+
+    const intervalId = setInterval(async () => {
+      attempts++;
+
+      try {
+        if (await closure()) {
+          clearInterval(intervalId);
+          resolve(true);
+        } else if (attempts >= maxAttempts) {
+          clearInterval(intervalId);
+          reject(new Error('Maximum attempts reached'));
+        }
+      } catch (error) {
+        clearInterval(intervalId);
+        reject(error);
+      }
+    }, pollTime);
+  });
+}
```

Comment:
wouldn't bother w/ set interval here honestly:
```suggestion
export async function tryUntilSuccess(
  closure: () => Promise<boolean>,
  pollTime: number,
  maxAttempts: number,
) {
  for (let attempts = 0; attempts < maxAttempts; attempts++) {
    await sleep(pollTime);
    if (await closure()) return;
  }
  
  throw new Error('Maximum attempts reached');
}
```

Reply:
haha yeah, just added that and pushed without seeing this, definitely over complicated (and it didn't work)